auth0/src/main/java/com/auth0/android/provider/WebAuthProvider.java

```diff
@@ -57,17 +60,86 @@
 
     private static final String TAG = WebAuthProvider.class.getName();
 
-    private static final String KEY_AUDIENCE = "audience";
-    private static final String KEY_SCOPE = "scope";
-    private static final String KEY_CONNECTION_SCOPE = "connection_scope";
-    private static final String SCOPE_TYPE_OPENID = "openid";
-    private static final String RESPONSE_TYPE_TOKEN = "token";
+    private static ResumableManager managerInstance;
 
-    private static OAuthManager managerInstance;
+    public static class LogoutBuilder {
 
+        private final Auth0 account;
+        private String scheme;
+        private CustomTabsOptions ctOptions;
+
+        LogoutBuilder(Auth0 account) {
+            this.account = account;
+
+            //Default values
+            this.scheme = "https";
+        }
+
+        /**
+         * When using a Custom Tabs compatible Browser, apply this customization options.
+         *
+         * @param options the Custom Tabs customization options
+         * @return the current builder instance
+         */
+        public LogoutBuilder withCustomTabsOptions(@NonNull CustomTabsOptions options) {
+            this.ctOptions = options;
+            return this;
+        }
+
+        /**
+         * Specify a custom Scheme to use on the Return To Uri. Default scheme is 'https'.
+         *
+         * @param scheme to use in the Callback Uri.
+         * @return the current builder instance
+         */
+        public LogoutBuilder withScheme(@NonNull String scheme) {
+            String lowerCase = scheme.toLowerCase();
+            if (!scheme.equals(lowerCase)) {
+                Log.w(TAG, "Please provide the scheme in lowercase and make sure it's the same configured in the intent filter. Android expects the scheme in lowercase");
+            }
+            this.scheme = scheme;
+            return this;
+        }
+
+        /**
+         * Request the user session to be cleared. When successful, the callback will get invoked
+         *
+         * @param context  to run the log out
+         * @param callback to invoke when log out is successful
+         */
+        public void start(Context context, BaseCallback<Void, Auth0Exception> callback) {
+            managerInstance = null;
+            if (account.getLogoutUrl() == null) {
+                Throwable cause = new IllegalArgumentException("Auth0 logout URL not properly set. This can be related to an invalid domain.");
```

Comment:
Unless this is common practice throughout the library, `IllegalArgumentException` says to me that I passed something into this method incorrectly so I would be looking at the `context` and `callback` instead of the `account` in the constructor. May something like:

> An Auth0 logout URL could not be created. Please check the account settings passed when initializing this class.

... or something to that effect. If it would only fail because of a missing/invalid domain, just say that explicitly. Assuming that this message is not shown to the user.

Reply:
Trying to find a justification on why this is checked here, I found that `Auth0` class already does check the domain has a proper URL syntax here https://github.com/auth0/Auth0.Android/blob/master/auth0/src/main/java/com/auth0/android/Auth0.java#L103. As you can see, it's throwing right there when you create a new instance of `Auth0` if the domain is not valid. I will remove the checks here and in the other authentication builder -> https://github.com/auth0/Auth0.Android/blob/master/auth0/src/main/java/com/auth0/android/provider/WebAuthProvider.java#L290